container/shim/src/fetchers/lassie.js

```diff
@@ -297,6 +307,31 @@ async function getRequestedBlockFromCar(streamIn, streamOut, cidObj, filename) {
   }
 }
 
+function generatePeersLassieUrl(knownPeers) {
+  const peerUrls = [];
+  if (knownPeers) {
+    const knownPeerList = [];
+    Object.values(knownPeers).forEach((peerList) => {
+      knownPeerList.push(...peerList);
+    });
+
+    knownPeerList.forEach((peer, idx) => {
+      const { peerID, multiaddr, protocol } = peer;
+
+      if (multiaddr && multiaddr.startsWith("http")) {
```

Comment:
Should this be checking `multiaddr` or `protocol` for `http`? 

The peer structure i'm working with is 
```
"knownPeers": {
    "web3.storage": [
      {
        "peerID": "QmUA9D3H7HeCYsirB3KmPSvZh3dNXMZas6Lwgr4fv1HTTp",
        "protocol": "http",
        "multiaddr": "/dns4/dag.w3s.link/tcp/443/https"
      }
    ]
  },
  ```

Reply:
The final url with the current logic will result in `"/dns4/dag.w3s.link/tcp/443/https/p2p/QmUA9D3H7HeCYsirB3KmPSvZh3dNXMZas6Lwgr4fv1HTTp+http"` which seems off.

Reply:
According to @hannahhoward's description, we pass the raw multiaddr only if the actual multiaddr starts with `http`